### PR TITLE
Template Store

### PR DESCRIPTION
**Pull request by Vijay-Giri**

### CODE REVIEW

File: editor/views.py [hunk 1]
```diff
@@ -36,3 +39,239 @@ def renderTemplate(request):
...
     return JsonResponse(data, safe=False)  # TOFIX: why not httpresponse
     # return HttpResponse(data, content_type='application/json')
+
+
+@csrf_exempt
+def template_view(request):
+    if request.method == "GET":
+        templates = Template.objects.all()
+        template_list = []
+        for template in templates:
```
Comment (wilspi) [inline editor/views.py]: serialize method in Template model

File: editor/models.py [hunk 2]
```diff
@@ -1,3 +1,79 @@
 from django.db import models
+import jsonfield
```
[the remark this answers is not in the document]
Comment (wilspi) [reply editor/models.py]: Remove jsonfield, this is deprecated, read https://pypi.org/project/django-jsonfield/
Also add packages to requirements.txt before using them

File: editor/models.py [hunk 3]
```diff
@@ -1,3 +1,79 @@
...
 
-# Create your models here.
+
+class Template(models.Model):
+
+    name = models.CharField(max_length=100)
+    default_version = models.IntegerField(null=True)
```
[the remark this answers is not in the document]
Comment (Vijay-Giri) [reply editor/models.py]: done

File: editor/frontend/src/templateScreen.js [hunk 4]
```diff
@@ -13,8 +13,8 @@ languages.forEach(lang => {
 import styles from './style/templateScreen.less';
 
 const defaultValue = {
-    template: `Hi {{name}},\nThis is a sample template.\nPaste your template here.`,
-    context: `{\n\t"name": "wilspi"\n}`,
+    template: `Hi {{name}}, {{version}},\nThis is a sample template.\nPaste your template here.`,
```
[the remark this answers is not in the document]
Comment (Vijay-Giri) [reply editor/frontend/src/templateScreen.js]: done

File: editor/models.py [hunk 5]
```diff
@@ -1,3 +1,79 @@
 from django.db import models
+import jsonfield
 
-# Create your models here.
+
+class Template(models.Model):
+
```
[the remark this answers is not in the document]
Comment (Vijay-Giri) [reply editor/models.py]: done

File: editor/views.py (hunk 1 above)
Comment (Vijay-Giri) [reply editor/views.py]: done by you

File: editor/views.py [hunk 6]
```diff
@@ -36,3 +42,149 @@ def renderTemplate(request):
...
+                default = 1
+                version = TemplateVersion.objects.get(
+                    pk=template.default_version_id
+                ).version
+
+            data = {"name": template.name, "version": version, "default": default}
+            template_list.append(data)
+        return JsonResponse(template_list, safe=False)
+
+    elif request.method == "POST":
```
[the remark this answers is not in the document]
Comment (wilspi) [reply editor/views.py]: 404

File: editor/views.py [hunk 7]
```diff
@@ -36,3 +42,149 @@ def renderTemplate(request):
 
     return JsonResponse(data, safe=False)  # TOFIX: why not httpresponse
     # return HttpResponse(data, content_type='application/json')
+
+
+@csrf_exempt
+def template_view(request):
+    if request.method == "GET":
+        offset = request.GET.get("offset")
```
[the remark this answers is not in the document]
Comment (Vijay-Giri) [reply editor/views.py]: done

File: editor/views.py [hunk 8]
```diff
@@ -36,3 +42,149 @@ def renderTemplate(request):
...
+
+        if limit is None:
+            limit = 100
+        else:
+            limit = int(limit)
+
+        templates = Template.objects.all()[offset : offset + limit]
+        template_list = []
+        for template in templates:
+            default = 0
```
[the remark this answers is not in the document]
Comment (Vijay-Giri) [reply editor/views.py]: done

File: editor/views.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (Vijay-Giri) [reply editor/views.py]: done

**Pull request merged**